--- a/zabbix.spec	2024-06-24 04:42:46.216086286 +0000
+++ b/zabbix.spec	2024-06-24 04:44:07.826703948 +0000
@@ -255,9 +255,16 @@
 Summary:		Zabbix server for MySQL or MariaDB database
 Group:			Applications/Internet
 Requires:		fping
+%if 0%{?rhel} >= 7
 Requires(post):		systemd
 Requires(preun):	systemd
 Requires(postun):	systemd
+%else
+Requires(post):		/sbin/chkconfig
+Requires(preun):	/sbin/chkconfig
+Requires(preun):	/sbin/service
+Requires(postun):	/sbin/service
+%endif
 Provides:		zabbix-server = %{version}-%{release}
 Provides:		zabbix-server-implementation = %{version}-%{release}
 %if 0%{?rhel} <= 8
@@ -272,9 +279,16 @@
 Summary:		Zabbix server for PostgresSQL database
 Group:			Applications/Internet
 Requires:		fping
+%if 0%{?rhel} >= 7
 Requires(post):		systemd
 Requires(preun):	systemd
 Requires(postun):	systemd
+%else
+Requires(post):		/sbin/chkconfig
+Requires(preun):	/sbin/chkconfig
+Requires(preun):	/sbin/service
+Requires(postun):	/sbin/service
+%endif
 Provides:		zabbix-server = %{version}-%{release}
 Provides:		zabbix-server-implementation = %{version}-%{release}
 %if 0%{?rhel} <= 8
@@ -511,6 +525,8 @@
 Zabbix frontend configuration for nginx (scl version)
 %endif
 # rhel == 7
+%endif
+# rhel >= 7
 
 %package java-gateway
 Summary:		Zabbix java gateway
@@ -536,8 +552,6 @@
 
 %description java-gateway
 Zabbix java gateway
-%endif
-# rhel >= 7
 
 %prep
 %setup0 -q -n %{name}-%{version}%{?alphatag}
@@ -823,8 +837,12 @@
 cat %{SOURCE3} | sed \
 	-e 's|COMPONENT|server|g' \
 	> $RPM_BUILD_ROOT%{_sysconfdir}/logrotate.d/zabbix-server
+%if 0%{?rhel} >= 7
 install -Dm 0644 -p %{SOURCE11} $RPM_BUILD_ROOT%{_unitdir}/zabbix-server.service
 install -Dm 0644 -p %{SOURCE15} $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d/zabbix-server.conf
+%else
+install -Dm 0755 -p %{SOURCE6} $RPM_BUILD_ROOT%{_sysconfdir}/init.d/zabbix-server
+%endif
 %endif
 
 
@@ -881,8 +899,12 @@
 cat src/zabbix_java/settings.sh | sed \
 	-e 's|^PID_FILE=.*|PID_FILE="/var/run/zabbix/zabbix_java.pid"|g' \
 	> $RPM_BUILD_ROOT%{_sysconfdir}/zabbix/zabbix_java_gateway.conf
+%if 0%{?rhel} >= 7
 install -Dm 0644 -p %{SOURCE13} $RPM_BUILD_ROOT%{_unitdir}/zabbix-java-gateway.service
 install -Dm 0644 -p %{SOURCE15} $RPM_BUILD_ROOT%{_prefix}/lib/tmpfiles.d/zabbix-java-gateway.conf
+%else
+install -Dm 0755 -p %{SOURCE4} $RPM_BUILD_ROOT%{_sysconfdir}/init.d/zabbix-java-gateway
+%endif
 %endif
 
 %clean
@@ -1179,8 +1201,12 @@
 %attr(0755,zabbix,zabbix) %dir %{_localstatedir}/log/zabbix
 %attr(0755,zabbix,zabbix) %dir %{_localstatedir}/run/zabbix
 %{_mandir}/man8/zabbix_server.8*
+%if 0%{?rhel} >= 7
 %{_unitdir}/zabbix-server.service
 %{_prefix}/lib/tmpfiles.d/zabbix-server.conf
+%else
+%{_sysconfdir}/init.d/zabbix-server
+%endif
 %{_sbindir}/zabbix_server_mysql
 
 %pre server-mysql
@@ -1224,8 +1250,12 @@
 %attr(0755,zabbix,zabbix) %dir %{_localstatedir}/log/zabbix
 %attr(0755,zabbix,zabbix) %dir %{_localstatedir}/run/zabbix
 %{_mandir}/man8/zabbix_server.8*
+%if 0%{?rhel} >= 7
 %{_unitdir}/zabbix-server.service
 %{_prefix}/lib/tmpfiles.d/zabbix-server.conf
+%else
+%{_sysconfdir}/init.d/zabbix-server
+%endif
 %{_sbindir}/zabbix_server_pgsql
 
 %pre server-pgsql
@@ -1398,8 +1428,12 @@
 %attr(0755,zabbix,zabbix) %dir %{_localstatedir}/run/zabbix
 %{_datadir}/zabbix-java-gateway
 %{_sbindir}/zabbix_java_gateway
+%if 0%{?rhel} >= 7
 %{_unitdir}/zabbix-java-gateway.service
 %{_prefix}/lib/tmpfiles.d/zabbix-java-gateway.conf
+%else
+%{_sysconfdir}/init.d/zabbix-java-gateway
+%endif
 %config(noreplace) %{_sysconfdir}/zabbix/zabbix_java_gateway_logback.xml
 
 %pre java-gateway
